src/clients/db/PatientRepository.ts

```diff
@@ -13,12 +13,13 @@ export class PatientRepository extends Repository<Patient> {
   }
 
   async search(text: string): Promise<Patient[]> {
+    const cleanText = text.replace(/[\\[\](){}?*]+/g, '')
```

Comment:
Maybe is better to use some stable package to search for char to leave out from a regexp? What do you think?

Reply:
Not sure what you mean by stable package...?

Do you mean something instead of the replace(regexp,'') method?

Reply:
You can use this https://www.npmjs.com/package/escape-string-regexp
It is ok to add a new dependency in this case.

Reply:
That particular package also removes hyphens, so we'd need to modify it.  

I'm new to TypeScript & node;  what would be the benefit of using a package in this instance?

Reply:
The benefit is that we are sure that we never miss any possible value. Since we will need this logic also on the server side and maybe in core package, it is better to use external package instead of copy-pasting the same code over several repositories.

Reply:
Ok.  What's the procedure for modifying existing packages?

Do I need to clone and modify the package you suggested (as we need to leave hyphens intact) and then install it, or do I add it to the master and edit it from within the project?

Reply:
It is easier like that. Just go in the root of this project and user yarn to add it as dependency
`yarn add escape-string-regexp` then go in this file and import it like `import escapeRegexp from 'escape-string-regexp'` and then you can use it in place of `.replace()`.

Reply:
For the hypens, they will be escaped, and that's what we need. Am I missing something?

Reply:
Hyphens are a relatively common character, especially in Western names.  Double-barrelled surnames names like John Cooper-Clarke are common in English, and hyphenated first names are common in French, for example;  'Jean-Pierre', or 'Anne-Marie' is not first-and-middle-name, it's just their first name.

Reply:
Hyphens are a relatively common character, especially in Western names.  Double-barrelled surnames names like John Cooper-Clarke are common in English, and hyphenated first names are common in French, for example;  'Jean-Pierre', or 'Anne-Marie' is not first-and-middle-name...it's just their first name.

Reply:
What i meant is that we just need to escape that chars, not to remove them. That's why I am suggesting to use that package.

Reply:
Ohhhhh.  *facepalm* . 

Yeah, I understand what you mean now.  Sorry, I was conflating escape with removal for some reason.  

Yes, that's a much neater approach, thank you.  I'll give it a go!

Reply:
No worries! I appreciate when there is a dialog on PR, since we it is important to not missing anything!
And, of course, thanks for your time and contribution!

Reply:
Ok, so I installed the package (though I still can't seem to get yarn to work, so used npm), and the new code seems to work fine:

```
async search(text: string): Promise<Patient[]> {
    const escapedString = escapeStringRegexp(text)
    return super.search({
      selector: {
        $or: [
          {
            fullName: {
              $regex: RegExp(escapedString, 'i'),
            },
          },
          {
            code: text,
          },
        ],
      },
    })
  }
```

However, the test suite that I wrote for the previous string cleaning method also needed to be updated, and it's failing the test:

```
describe('search', () => {
    afterEach(async () => {
      await removeAllDocs()
    })

    it('should escape all special chars from search text', async () => {
      await patients.put({ _id: 'id9999', code: 'P00001', fullName: 'test -]?}(){*[\\$+.^test' })

      const result = await PatientRepository.search(
        'test \\-\\]\\?\\}\\(\\)\\{\\*\\[\\\\\\$\\+\\.\\^test',
      )

      expect(result).toHaveLength(1)
      expect(result[0].id).toEqual('id9999')
    })

```

It's coming back as result length = 0, so it's not finding a match.

Am I misunderstanding what the result string should be? 

Or perhaps something to do with the method of the new package inserting a \unnnn escape:

> Use a simple backslash escape when it’s always valid, and a \unnnn escape when the simpler form would be disallowed by Unicode patterns’ stricter grammar.

As I understand it, the escaped string should be the same as the input string, but with the special chars escaped with a backslash...or am I trying to read backslashes that aren't there, because they were ignored?

Reply:
Am I perhaps overcomplicating this, and only need to test that the input and output strings match directly?